horovod/torch/__init__.py

```diff
@@ -72,23 +79,42 @@ def _register_hooks(self):
                     grad_acc.register_hook(self._make_hook(p))
                     self._grad_accs.append(grad_acc)
 
+    def _all_reduce_grad(self, p):
+        name = self._parameter_names.get(p)
+        tensor = p.grad.data
+        tensor_compressed, ctx = self._compression.compress(tensor)
+
+        handle = allreduce_async_(tensor_compressed, average=True, name=name)
+        return handle, ctx
+
     def _make_hook(self, p):
         def hook(*ignore):
-            assert p not in self._handles
+            if p in self._handles and self._handles[p][0] is not None:
+                if self._parameter_update_delay[p] <= 0:
+                    raise AssertionError(
+                        "Gradients were accumulated twice without a "
```

Comment:
With other comments in mind, we can say something like `Gradients were computed more than backward_passes_per_step times without call to step().  Please increase backward_passes_per_step to accumulate gradients locally.`